app/models/course_year.rb

```diff
@@ -17,4 +18,32 @@ def course_modules_in_order
     preloaded_modules = course_modules.includes(:previous_module, :next_module)
     elements_in_order(elements: preloaded_modules, previous_method_name: :previous_module)
   end
+
+  def modules_with_progress(user)
+    modules_in_order = course_modules_in_order
+    ect_profile = user&.early_career_teacher_profile
+    return modules_in_order unless ect_profile
+
+    set_user_course_module_progresses(get_user_lessons_and_progresses(ect_profile))
+  end
+
+private
+
+  def set_user_course_module_progresses(course_lessons)
+    course_modules.map do |course_module|
```

Comment:
Probably want `modules_in_order` here - and I would expect them to be passed to the method to avoid re-computing them.